### PR TITLE
Bartender view

### PR DESCRIPTION
**Pull request by kprakobkit**
Might be overkill but i took out an OpenEntryQueue component that takes a filter predicate and displayMax.

### CODE REVIEW

File: src/components/Bartender.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    this.props.loadOrders();
+  }
+
+  render() {
+    const isOpen = _.pathEq(['entry', 'status'], constants.OPEN);
+    const isDrink = _.pathEq(['entry', 'type'], constants.DRINK);
+
+    return OpenEntryQueue({
+      orders: this.props.orders,
+      displayMax: 6,
```
Comment (achiu8) [inline src/components/Bartender.js:27]: Do we think that bartender and chef will want to view different amounts?

File: src/components/Bartender.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  };
+}
+
+class Bartender extends Component {
+  componentWillMount() {
+    this.props.loadOrders();
+  }
+
+  render() {
+    const isOpen = _.pathEq(['entry', 'status'], constants.OPEN);
```
Comment (achiu8) [inline src/components/Bartender.js]: Should the open status check be built into the `OpenEntryQueue` component, given its name?

File: src/components/Bartender.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import { Component, PropTypes, DOM as dom, createFactory } from 'react';
+import { connect } from 'react-redux';
+import actions from '../action_creators';
+import constants from '../constants';
+import moment from 'moment';
+import OpenEntryQueueContainer from './OpenEntryQueue';
+import _ from 'ramda';
+
+const OpenEntryQueue = createFactory(OpenEntryQueueContainer);
```
Comment (achiu8) [inline src/components/Bartender.js]: `OpenEntryQueueContainer` should be `OpenEntryQueueComponent` right?

**Comment by achiu8** [comment]
Looks like `Chef` and `Bartender` are pretty much the same for now, besides the type of entry they display. Do we think that the views will differ more meaningfully in the future? Otherwise, I think we could probably consolidate and abstract out a `Kitchen` component or something.

File: src/components/Bartender.js (hunk 2 above)
Comment (kprakobkit) [inline src/components/Bartender.js]: Yeah was thinking the same. Will do.

**Comment by kprakobkit** [comment]
What would be a kitchen component? Not sure how much more they'll differ, will need to check with PO (product owner).

File: src/components/Bartender.js (hunk 1 above)
Comment (kprakobkit) [inline src/components/Bartender.js:27]: It would make sense, since the bar should be able to process more at a time...? Will double check with PO.

**Comment by achiu8** [comment]
So I was thinking that `Kitchen` would just be the same as `Chef` or `Bartender`, with the following changes to `render`:

```
const filterType = this.props.route.path === '/chef' ? constants.FOOD : constants.DRINK;
return OpenEntryQueue({
  orders: this.props.orders,
  displayMax: 6,
  changeEntryStatus: this.props.changeEntryStatus,
  filterPredicate: _.pathEq(['entry', 'type'], filterType)
});
```

Then could render this same component for both the `/chef` and `/bartender` routes, and save some redundancy. Although it does feel a little weird.. just a thought.

**Comment by kprakobkit** [comment]
I see. I think it would make it less clear that bartender only displays drinks because there is now logic in the component that determines the filter type as opposed to explicitly declaring filter type for a Bartender/Chef component. Also would be kinda weird to test route dependency.

**Comment by achiu8** [comment]
Makes sense, let's leave it.

**Pull request merged**